CMakeLists.txt

```diff
@@ -1828,8 +1828,15 @@ endif()
 # Generate and use MCUboot related artifacts as needed.
 if(CONFIG_BOOTLOADER_MCUBOOT)
   get_target_property(signing_script zephyr_property_target SIGNING_SCRIPT)
+
   if(NOT signing_script)
-    set_target_properties(zephyr_property_target PROPERTIES SIGNING_SCRIPT ${CMAKE_CURRENT_LIST_DIR}/cmake/mcuboot.cmake)
+    zephyr_get(signing_script VAR SIGNING_SCRIPT SYSBUILD)
```

Comment:
Should this be moved outside of `if(CONFIG_BOOTLOADER_MCUBOOT)`? I would expect this new cache variable to be as generic as the old target property.

Also, I suggest the following cleanup (feel free to ignore):
```cmake
# Include signing script, if set either in CMake cache or as a target property.
get_target_property(signing_script zephyr_property_target SIGNING_SCRIPT)
if(NOT signing_script)
  zephyr_get(signing_script VAR SIGNING_SCRIPT SYSBUILD)
endif()
if(NOT signing_script AND CONFIG_BOOTLOADER_MCUBOOT)
  # Default script for MCUboot related artifacts.
  set(signing_script ${CMAKE_CURRENT_LIST_DIR}/cmake/mcuboot.cmake)
endif()
```